src/Interpreters/InsertDependenciesBuilder.h

```diff
@@ -43,13 +43,32 @@ class InsertDependenciesBuilder : public std::enable_shared_from_this<InsertDepe
         bool operator==(const StorageIDPrivate & other) const;
     };
 
+    class StorageIDPrivateSet
+    {
+    private:
+        std::set<StorageIDPrivate> visited;
+    public:
+        void insert(const StorageIDPrivate & id) { visited.insert(id); }
+        /// std::set::contains() uses the operator< and if it is inconsistent
+        /// with the operator== then it will not work correctly.
+        /// Take a look at the comment for operator== in the StorageID itself.
+        bool contains(const StorageIDPrivate & id) const
+        {
+            for (const auto & x : visited)
+                if (x == id)
+                    return true;
+            return false;
+        }
+        void erase(const StorageIDPrivate & id) { visited.erase(id); }
```

Comment:
Hmm, it seems this one is also not consistent with contains and also uses `operator<`. Should we just replace everything with std::vector? Or replace with std::unordered_map and calculate a hash of StorageID?

Reply:
Let's just use std::unoredered_set.